### PR TITLE
Stabilize is_partitioned

### LINKED ISSUE #62544: Tracking issue for Iterator::is_partitioned
```rust
/// Checks if the elements of this iterator are partitioned according to the given predicate,
/// such that all those that return `true` precede all those that return `false`.
fn is_partitioned<P>(mut self, mut predicate: P) -> bool
where
    Self: Sized,
    P: FnMut(Self::Item) -> bool,
```

`feature = "iter_is_partitioned"`

ref: #62278

### PR DESCRIPTION
**Pull request by TheWastl**
Closes #62544

### CODE REVIEW

File: library/core/src/iter/traits/iterator.rs [hunk 1]
```diff
@@ -1802,12 +1802,10 @@ pub trait Iterator {
     /// # Examples
     ///
     /// ```
-    /// #![feature(iter_is_partitioned)]
-    ///
     /// assert!("Iterator".chars().is_partitioned(char::is_uppercase));
     /// assert!(!"IntoIterator".chars().is_partitioned(char::is_uppercase));
     /// ```
-    #[unstable(feature = "iter_is_partitioned", reason = "new API", issue = "62544")]
+    #[stable(feature = "iter_is_partitioned", since = "1.49.0")]
```
Comment (m-ou-se) [inline library/core/src/iter/traits/iterator.rs:1808]: 1.49 is already in beta and 1.50 is going to be in beta soon, so this should probably be 1.51.

**Comment by m-ou-se** [comment]
@rfcbot merge

File: library/core/src/iter/traits/iterator.rs (hunk 1 above)
Comment (TheWastl) [reply library/core/src/iter/traits/iterator.rs]: Changed

**Comment by BurntSushi** [comment]
I've looked at the various linked issues but didn't see any motivating use cases for this routine. cc @cuviper

**Comment by cuviper** [comment]
My own motivation was just that [someone asked for partitioning like C++ `<algorithm>`](https://users.rust-lang.org/t/include-algorithm/29727), and it was easy enough to emulate [`std::is_partitioned`](https://en.cppreference.com/w/cpp/algorithm/is_partitioned) at the same time. I don't personally need it though.

**Comment by TheWastl** [comment]
I don't personally need it either. My motivation for this PR was also someone else. They were wondering why it was not stabilized, as there had been no activity on the tracking issue for almost 1.5 years. I've told the person to comment here.

**Comment by m-ou-se** [comment]
Thinking a bit more about this, I suppose that pretty much all use cases of this function are also covered by `is_sorted_by_key` .

**Comment by m-ou-se** [comment]
I've added the unresolved question to the tracking issue. Cancelling the FCP for now, as we first need to discuss/resolve that.

@rfcbot cancel

**Comment by m-ou-se** [comment]
Let's continue the discussion on the tracking issue. Thanks!

**Pull request closed without merging**